src/examples/go_monorepo_workflows.yml

```diff
@@ -0,0 +1,187 @@
+description: |
+  This example shows:
+   - Specific version of the isopod.
+   - this is for go application
+   - custom quality checks
+usage:
+  version: 2.1
+
+  orbs:
+    go: circleci/go@1.7.1
+    ric-orb: ricardo/ric-orb@dev:e4c7a02
+
+  cfg-isopod-version: &cfg-isopod-version
+    isopod_version: 0.29.1
+  cfg-go-version: &cfg-go-version
+    tag: '1.17'
+
+  parameters:
+    api:
+      type: boolean
+      default: false
+    worker:
+      type: boolean
+      default: false
+    common:
+      type: boolean
+      default: false
+
+  workflows:
+    version: 2
+    api:
+      when:
+        or:
+          - << pipeline.parameters.api >>
+          - << pipeline.parameters.common >>
+      jobs:
+        - quality-gate:
+            context: dev
+            appname: api
+        - request_branch_deploy_dev:
+            type: approval
+            name: request-deploy
+            filters:
+              branches:
+                ignore:
+                  - master
+        - ric-orb/build_push_image:
+            <<: *cfg-isopod-version
+            context: dev
+            work_dir: './api'
+            name: docker-branch
+            requires:
+              - request-deploy
+              - quality-gate
+        - ric-orb/deploy_job:
+            <<: *cfg-isopod-version
+            env: dev
+            context: dev
+            name: deploy-dev
+            work_dir: './api'
+            requires:
+              - docker-branch
+        #master branch
+        - ric-orb/build_push_image:
+            <<: *cfg-isopod-version
+            context: prod
+            prebuild_steps:
+              - attach_workspace:
+                  at: .
+            work_dir: './api'
+            name: docker-master
+            filters:
+              branches:
+                only:
+                  - master
+            requires:
+              - quality-gate
```

Comment:
is the resulting docker image depending on context? otherwise this is producing the same result as the job for the branch!
if its the same the workflow can be simplified a lot!

Reply:
I've checked env vars and it seems that they are the same in both context, but this way we would need to approve deployment of master branch. I would stick to current flow.